tests/test_export_db.py

```diff
@@ -0,0 +1,229 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+
+import os
+import sys
+import argparse
+import logging
+import pytest
+
+from sqlalchemy.orm import Query
+from sqlalchemy.ext.declarative import declarative_base
+
+sys.path.append("../tuna")
+sys.path.append("tuna")
+
+this_path = os.path.dirname(__file__)
+
+from tuna.miopen.subcmd.export_db import (
+    arg_export_db, get_filename, get_base_query, get_fdb_query, get_pdb_query,
+    get_fdb_alg_lists, build_miopen_fdb, write_fdb, export_fdb,
+    build_miopen_kdb, write_kdb, export_kdb, create_sqlite_tables, get_cfg_dict)
+from tuna.utils.db_utility import DB_Type
+from tuna.miopen.db.tables import MIOpenDBTables, ConfigType
+from tuna.dbBase.sql_alchemy import DbSession
+from tuna.miopen.db.find_db import ConvolutionFindDB
+from utils import add_test_session, DummyArgs
+
+
+@pytest.fixture
+def mock_args():
+  args = argparse.Namespace()
+  args.golden_v = None
+  args.arch = "arch"
+  args.num_cu = 64
+  args.opencl = "1"
+  args.config_tag = None
+  args.filename = None
+  return args
+
+
+#testing fdb and pdb functions
+def build_fdb_entry(session_id):
+  fdb_entry = ConvolutionFindDB()
+  fdb_entry.config = 1
+  fdb_entry.solver = 1
+  fdb_entry.session = session_id
+  fdb_entry.opencl = False
+
+  fdb_entry.fdb_key = 'key'
+  fdb_entry.alg_lib = 'Test'
+  fdb_entry.params = 'param'
+  fdb_entry.workspace_sz = 0
+  fdb_entry.valid = True
+  fdb_entry.kernel_time = 11111
+  fdb_entry.kernel_group = 1
+
+  return fdb_entry
+
+
+session_id = add_test_session()
+fdb_entry = build_fdb_entry(session_id)
+args = DummyArgs()
+args.session_id = session_id
+args.config_type = ConfigType.convolution
+dbt = MIOpenDBTables(session_id=args.session_id, config_type=args.config_type)
+logger = logging.getLogger("test_logger")
+
+
+def test_get_file():
+  arch = "gfx900"
+  num_cu = 64
+  filename = None
+  ocl = True
+  db_type = DB_Type.FIND_DB
+
+  expeted_filename = "tuna_1.0.0/gfx900_64.OpenCL.fdb.txt"
+  actual_filename = get_filename(arch, num_cu, filename, ocl, db_type)
+
+  assert actual_filename == expeted_filename, f"expected {expeted_filename}, but got {actual_filename}"
+
+
+with DbSession() as session:
+  session.add(fdb_entry)
+  session.commit()
+
+with DbSession() as session:
```

Comment:
instead of nesting the test calls inside the with statement, please create a session and pass it around to the tests. You can then do session.close or something like that